server/routes/searchComments.ts

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable quotes */
+import { Request, Response, NextFunction } from 'express';
+import { QueryTypes } from 'sequelize';
+import lookupCommunityIsVisibleToUser from '../util/lookupCommunityIsVisibleToUser';
+import { DB } from '../database';
+
+const Errors = {
+  UnexpectedError: 'Unexpected error',
+  QueryMissing: 'Must enter query to begin searching',
+  QueryTooShort: 'Query must be at least 4 characters',
+  NoCommunity: 'Title search must be community scoped'
+};
+
+const searchComments = async (
+  models: DB,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  let replacements = {};
+
+  if (!req.query.search) {
+    return next(new Error(Errors.QueryMissing));
+  }
+  if (req.query.search.length < 4) {
+    return next(new Error(Errors.QueryTooShort));
+  }
+
+  // Community-scoped search
+  let communityOptions = '';
+  if (req.query.chain || req.query.community) {
+    const [chain, community, error] = await lookupCommunityIsVisibleToUser(
+      models,
+      req.query,
+      req.user
+    );
+    if (error) return next(new Error(error));
+
+    // set up query parameters
+    communityOptions = community
+      ? `AND "OffchainComments".community = :community`
+      : `AND "OffchainComments".chain = :chain `;
+    replacements = community
+      ? { community: community.id }
+      : { chain: chain.id };
+  }
+
+  const sort = req.query.sort === 'Newest'
+    ? 'ORDER BY "OffchainComments".created_at DESC'
+    : req.query.sort === 'Oldest'
+    ? 'ORDER BY "OffchainComments".created_at ASC'
+    : 'ORDER BY rank DESC'
+
+  replacements['searchTerm'] = req.query.search;
+  replacements['limit'] = 50; // must be same as SEARCH_PAGE_SIZE on frontend
+
+  // query for both threads and comments, and then execute a union and keep only the most recent :limit
+  let comments;
+  try {
+    comments = await models.sequelize.query(
```

Comment:
I'm ever so slightly concerned about constructing this query in this way (rather than using the sequelize orm syntax) but I understand the complexity of this query to be necessary. I need @jnaviask to approve this being we had SQL injection attacks so recently, gotta make sure this passes a few other eyes.

Reply:
Good idea to get it checked, but the structure of the query is the same as it was before in Search 1.0/2.0 - I've only used some new functions for parsing the query and actually executing the search + filtering.

Reply:
I'm not sure we can do `websearch_to_tsquery` in sequelize...